src/RouteRegistrar.php

```diff
@@ -85,15 +95,20 @@ public function registerClass(string $class): void
 
     protected function fullQualifiedClassNameFromFile(SplFileInfo $file): string
     {
-        $class = trim(Str::replaceFirst($this->basePath, '', $file->getRealPath()), DIRECTORY_SEPARATOR);
-
-        $class = str_replace(
-            [DIRECTORY_SEPARATOR, 'App\\'],
-            ['\\', app()->getNamespace()],
-            ucfirst(Str::replaceLast('.php', '', $class))
-        );
+        $basePath = str_replace(['/', '\\'], DIRECTORY_SEPARATOR, $this->basePath);
+        $class = trim(Str::replaceFirst($basePath, '', $file->getRealPath()), DIRECTORY_SEPARATOR);
+
+        if ($this->hasNamespaceKey) {
+            $class = '\\'.ucfirst(Str::replaceLast('.php', '', $class));
+        } else {
+            $class = str_replace(
```

Comment:
It's not immediately clear what happens here. Please put this in a function with a descriptive name.